### PR TITLE
test: Use a Q35 machine by default in tests

### PR DESCRIPTION
**Pull request by mvollmer**
This is a newer chipset and creating a virtual machine in the UI or
with virt-install will very likely use it. Thus, let's also use it in
our tests.

### CODE REVIEW

**Comment by martinpitt** [comment]
#1590 landed, so this now needs a rebase.

**Comment by mvollmer** [comment]
Hmm, q35 in rhel-8-10 seems to not allow enough hotplugging for our tests.

**Comment by mvollmer** [comment]
Pixel test flake is being fixed in #1614

**Comment by mvollmer** [comment]
> Pixel test flake is being fixed in #1614

Turns out this is something new with linux2022, actually.

File: test/check-machines-disks [hunk 1]
```diff
@@ -82,7 +82,11 @@ class TestMachinesDisks(machineslib.VirtualMachinesCase):
             else:
                 b.wait_not_present(f"#vm-{name}-disks-{target}-edit-dialog")
 
-        self.createVm("subVmTest1")
+        if "rhel-8" in m.image:
+            # q35 in rhel-8 can't hotplug
+            self.createVm("subVmTest1", os="linux2016")
```
Comment (martinpitt) [inline test/check-machines-disks:87]: This is done several times -- perhaps factor this out into setUp(), create some `create_options` dict (empty by default, that option on rhel8), and call `self.createVm("subVmTest1", **create_options)`?

That should also apply to centos-8-stream, no?

**Review by martinpitt — COMMENTED** [review]
Ooh, really nice cleanup of the xml parsing mess in the tests! :hugs: 

You may need a force-push anyway to get around that hanging TF test. One suggestion for code cleanup.

File: test/check-machines-disks (hunk 1 above)
Comment (mvollmer) [reply test/check-machines-disks:87]: Yep. I might try sticking with linux2016 on rhel-8 in general, and then bumping it up to linux2022 in specific tests explicitly.

Comment (mvollmer) [reply test/check-machines-disks]: Oh, this just works.

Comment (mvollmer) [reply test/check-machines-disks]: > That should also apply to centos-8-stream, no?

We don't test on centos-8-stream here, do we?

Comment (martinpitt) [reply test/check-machines-disks]: Right, that overlapped with yesterday's c8s cleanup.

**Review by martinpitt — APPROVED** [review]
Yay!

**Pull request merged**